### DISCUSSION TITLE
error handling

### DISCUSSION THREAD

**Discussion by edw19** [urql-graphql/urql#1861] [Q&A] [answered]
hi !! I have new question about how urql works
my repo:  https://github.com/edw19/next-urql-example

i'm forzed error in this mutation for handle the error in client 
![image](https://user-images.githubusercontent.com/47750859/129598050-485529b5-9848-43ef-b230-ed5abe60fa91.png)

this is my hook in client, in createProduct function handle the creating the product but i realized than mutation with error it is completed
![image](https://user-images.githubusercontent.com/47750859/129598433-2b88e999-0640-4bd0-9150-77c5818f746b.png)

![image](https://user-images.githubusercontent.com/47750859/129598903-d14432e8-f271-43ac-9fcc-d7be454dd08a.png)

this is the expected behavior
![image](https://user-images.githubusercontent.com/47750859/129599509-4a27b71d-a9c8-4811-93ba-40edee3b7833.png)

**Comment by kitten** [comment, marked as answer]
I'm not really sure what answer you're expecting here 😅 This is all documented here: https://formidable.com/open-source/urql/docs/basics/react-preact/#handling-mutation-errors

**Reply by edw19** [reply]
I think my problem is in the part of the documentation that says that the "It's worth noting that the promise we receive when calling the execute function will never reject. Instead it will always return a promise that resolves to a result."
I was trying to handle the error with a try catch.
Sorry for the next time I will read the documentation better
thanks for the reply

**Reply by kitten** [reply]
yea no worries! also make sure to use `.then` or `awaut` on the execute function for the mutation if you need the result right there.

It's worth noting that if you have a promise in any part of urql, it'll never reject. We never outright reject promises because there are different kinds of errors and what they mean and a try-catch is often too restrictive by default when there's "half-errored" states that are possible as well
